renku/ui/service/serializers/templates.py

```diff
@@ -17,30 +17,51 @@
 
 from urllib.parse import urlparse
 
-from marshmallow import Schema, ValidationError, fields, post_load, pre_load
-from yagup import GitURL
+import yagup
+from marshmallow import Schema, ValidationError, fields, post_load, pre_load, validates
 from yagup.exceptions import InvalidURL
 
 from renku.core.util.os import normalize_to_ascii
 from renku.domain_model.dataset import ImageObjectRequestJson
 from renku.ui.service.config import TEMPLATE_CLONE_DEPTH_DEFAULT
-from renku.ui.service.serializers.cache import ProjectCloneContext, RepositoryCloneRequest
+from renku.ui.service.errors import UserRepoUrlInvalidError
+from renku.ui.service.serializers.common import GitUrlResponseMixin
 from renku.ui.service.serializers.rpc import JsonRPCResponse
 from renku.ui.service.utils import normalize_git_url
 
 
-class ManifestTemplatesRequest(RepositoryCloneRequest):
+class ManifestTemplatesRequest(Schema):
     """Request schema for listing manifest templates."""
 
-    url = fields.String(required=True)
+    template_git_url = fields.String(required=True, metadata={"description": "Template git repository url."})
     depth = fields.Integer(load_default=TEMPLATE_CLONE_DEPTH_DEFAULT)
+    branch = fields.String(load_default=None, metadata={"description": "Remote git branch (or tag or commit SHA)."})
+
+    @pre_load
+    def set_fields(self, data, **_):
+        """Set `branch` field from `ref` if present and set template url."""
+        if "ref" in data and not data.get("branch"):
+            # Backward compatibility: branch and ref were both used. Let's keep branch as the exposed field
+            # even if internally it gets converted to "ref" later.
```

Comment:
I loooked through all versins of `templates.read_manifest` on swagger and not a single one has `branch`. Where did we use `branch` before?